packages/manager/modules/pci/src/projects/project/instances/instances.controller.js

```diff
@@ -34,6 +40,77 @@ export default class CloudProjectComputeInfrastructureListCtrl {
     this.checkHelpDisplay();
 
     this.criteria = getCriteria('id', this.instanceId);
+
+    this.fetchInstances();
+  }
+
+  fetchInstances() {
+    this.loading = true;
+    return this.$q
+      .all({
+        instances: this.PciProjectsProjectInstanceService.getAll(
+          this.projectId,
+          this.customerRegions,
+        ),
+        floatingIps: this.getFloatingIps(),
+      })
+      .then(({ instances, floatingIps }) => {
+        const updatedInstances = instances.map((instance) => ({
+          ...instance,
+          floatingIp: floatingIps.find(
+            (floatingIp) => floatingIp?.associatedEntity?.id === instance.id,
+          ),
+        }));
+        return this.$q.all(
+          updatedInstances.map((instance) => {
+            return this.PciProjectsProjectInstanceService.getInstanceFlavor(
+              this.projectId,
+              instance,
+            ).then((flavor) => {
+              return new Instance({
+                ...instance,
+                flavor,
+              });
+            });
+          }),
+        );
+      })
+      .then((instances) => {
+        if (instances.length === 0) {
+          this.$state.go('pci.projects.project.instances.onboarding');
+        } else {
+          this.instances = instances;
+          this.instancesRegions = Array.from(
+            new Set(instances.map(({ region }) => region)),
+          );
+          this.loading = false;
+          this.pollInstances();
+        }
+      });
+  }
+
+  pollInstances() {
+    this.instances.forEach((instance) => {
+      POLLER_INSTANCES.filter((poller) =>
+        poller.needsPolling(instance),
+      ).forEach((poller) => {
+        const endPointUrl = `/cloud/project/${this.projectId}/instance/${instance.id}?region=${instance.region}`;
+        this.Poller.poll(endPointUrl, null, {
+          interval: 5000,
+          successRule: poller.isPolled,
```

Comment:
and here `successRule: () => poller.isPolled(instance)`

Reply:
This one won't work because you need to get the instance from the success call, I will use old instance when possible